### PR TITLE
feat: add otlp trace provider

### PR DESCRIPTION
**Pull request by matoous**
This PR adds support for `otlp` traces exporter usable with providers such as NewRelic, Honeycomb, or AWS X-Ray. Since the support for tracing is already in place the only necessary change was to include yet another library for tracing and add appropriate tests. `otel` tracing has plenty of configuration options that I would like to address in a separate PR ones the basic for `otlp` exporter is tested.

## Related Issue or Design Document

* https://github.com/ory/hydra/issues/2130
* https://github.com/ory/hydra/issues/2597

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: tracing/tracer.go [hunk 1]
```diff
@@ -164,6 +172,43 @@ func (t *Tracer) setup() error {
...
+				semconv.ServiceNameKey.String(serviceName),
+			),
+		)
+		if err != nil {
+			return errors.Wrap(err, "new otlp resource")
+		}
+
+		exporter, err := otlptracehttp.New(ctx)
+		if err != nil {
+			return errors.Wrap(err, "new otlp resource")
```
Comment (svrakitin) [inline tracing/tracer.go:193]:
```suggestion
			return errors.Wrap(err, "new otlp exporter")
```

File: tracing/tracer.go [hunk 2]
```diff
@@ -164,6 +172,43 @@ func (t *Tracer) setup() error {
 		opentracing.SetGlobalTracer(t.tracer)
 
 		t.l.Infof("Instana tracer configured!")
+	case "otlp":
+		ctx := context.Background()
+		var serviceName = os.Getenv("OTLP_SERVICE_NAME")
```
Comment (svrakitin) [inline tracing/tracer.go:177]:
```suggestion
		var serviceName = os.Getenv("OTEL_SERVICE_NAME")
```

File: tracing/tracer.go [hunk 3]
```diff
@@ -164,6 +172,43 @@ func (t *Tracer) setup() error {
 		opentracing.SetGlobalTracer(t.tracer)
 
 		t.l.Infof("Instana tracer configured!")
+	case "otlp":
```
Comment (svrakitin) [inline tracing/tracer.go:175]: Maybe like this?
```suggestion
	case "otel":
```

File: tracing/tracer.go [hunk 4]
```diff
@@ -164,6 +172,43 @@ func (t *Tracer) setup() error {
...
+			serviceName = t.Config.ServiceName
+		}
+
+		res, err := resource.New(ctx,
+			resource.WithAttributes(
+				semconv.ServiceNameKey.String(serviceName),
+			),
+		)
+		if err != nil {
+			return errors.Wrap(err, "new otlp resource")
```
Comment (svrakitin) [inline tracing/tracer.go:188]:
```suggestion
			return errors.Wrap(err, "new otel resource")
```

**Review by aeneasr — APPROVED** [review]

**Pull request merged**